### PR TITLE
Add more example code standards

### LINKED ISSUE #983: Improve PyGMT documentation
We have to decide whether we want to add quotes to all arguments, e.g., `'05m'` or `05m`. Originally posted in https://github.com/GenericMappingTools/pygmt/pull/976#discussion_r584484932. 

We use quotes for many arguments, e.g.,
https://github.com/GenericMappingTools/pygmt/blob/afe9293d8d6705587be9982909820d9ebf5e75eb/pygmt/datasets/earth_relief.py#L34-L39

Meanwhile, we don't use quotes in some cases, e.g.,
https://github.com/GenericMappingTools/pygmt/blob/afe9293d8d6705587be9982909820d9ebf5e75eb/pygmt/datasets/earth_relief.py#L44-L45

We could refer to other projects posted in https://github.com/GenericMappingTools/pygmt/pull/976#discussion_r584759875
> Just checked other projects, numpy (https://numpy.org/doc/stable/reference/generated/numpy.datetime_as_string.html#numpy.datetime_as_string), pandas (https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html) and matplotlib (https://matplotlib.org/stable/api/legend_api.html) use `'string'`, not `` ``'string'`` ``.

### PR DESCRIPTION
**Pull request by core-man**
**Description of proposed changes**

A code standard about string value is added in [CONTRIBUTING.md](https://github.com/GenericMappingTools/pygmt/blob/master/CONTRIBUTING.md) (Originally posted in https://github.com/GenericMappingTools/pygmt/issues/983#issue-818987282. See also https://github.com/GenericMappingTools/pygmt/issues/983#issuecomment-806716900 and https://github.com/GenericMappingTools/pygmt/issues/983#issuecomment-807224037):

- Double-quotes are used to indicate a string value instead of single-quotes. Example: "05m".
- An argument with a string value set should be added \`\` on both sides of the expressions. Example: \`\`style="v"\`\` results in ``style="v"``.

Fixes #983 

**Reminders**

- [x] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Review by meghanrjones — APPROVED** [review]
This guideline seems fine to me, but we can leave it open for a bit longer to see if anyone else has other preferences.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -188,6 +188,9 @@ When editing documentation, use the following standards to demonstrate the examp
    Example: `**+l**\ *label*` results in **+l***label*.
 3. Optional arguments are placed wrapped with [ ] (square brackets).
 4. Arguments that are mutually exclusive are separated with a | (bar) to denote "or".
+5. Double-quotes are used to indicate a string value. Example: "05m". An argument
+   with a string value set should be added \`\` on both sides of the expressions.
+   Example: \`\`style="v"\`\` results in ``style="v"``.
```
Comment (seisman) [inline CONTRIBUTING.md:193]: This file uses markdown syntax:
```suggestion
   Example: \`\`style="v"\`\` results in `style="v"`.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -188,6 +188,9 @@ When editing documentation, use the following standards to demonstrate the examp
    Example: `**+l**\ *label*` results in **+l***label*.
 3. Optional arguments are placed wrapped with [ ] (square brackets).
 4. Arguments that are mutually exclusive are separated with a | (bar) to denote "or".
+5. Double-quotes are used to indicate a string value. Example: "05m". An argument
```
Comment (seisman) [inline CONTRIBUTING.md:191]: As mentioned in https://github.com/GenericMappingTools/pygmt/issues/983#issuecomment-807224037:

> Another alternative way is using bold text, for example, **05m**, as bold text means literal arguments in both GMT and PyGMT documentation.

This standard recommend double quotes for strings ("05m"), but some GMT arguments use bold text. For example, the `verbose` parameter can accept strings like "w", "d" and others, and we use bold strings following the GMT convention (see https://www.pygmt.org/dev/api/generated/pygmt.Figure.basemap.html#pygmt.Figure.basemap). 

Will it cause more confusion?

Comment (meghanrjones) [reply CONTRIBUTING.md:191]: GMT and PyGMT both use bold for literal arguments in the html documentation, but GMT uses different formatting for the command line documentation. I think this is the main downside of using the formatting standards from GMT's RST documentation in the python docstrings and where using bold inside strings could get confusing. For example, here is the python console help for the `verbose` parameter in Figure.basemap()
image
relative to the command line documentation for the **-V** option in GMT's basemap module:
image

Comment (core-man) [reply CONTRIBUTING.md:191]: Actually, I am fine with bold strings. I think Python users know whether it is a string and how to indicate a string value.

Comment (seisman) [reply CONTRIBUTING.md:191]: I'm still hesitate to adopt either style: **bold** or "double-quotes".

Comment (meghanrjones) [reply CONTRIBUTING.md:191]: > I'm still hesitant to adopt either style: **bold** or "double-quotes".

Why? Or what do you prefer?

Comment (seisman) [reply CONTRIBUTING.md:191]: We have to make a choice. Let's adopt **bold** texts.

Comment (meghanrjones) [reply CONTRIBUTING.md:191]: I understand using bold text for stand-along strings (for example, the list of verbose options that you originally shared), but when it is within a code block I think it needs to be double quotes. For example, do a search for single quotes in this latest commit https://github.com/GenericMappingTools/pygmt/commit/e131d26f9666bb28aa11d938f232f30036791976 and you will see that most are part of a code block inside docstrings.

Comment (seisman) [reply CONTRIBUTING.md:191]: > when it is within a code block I think it needs to be double quotes.

Yes, I agree.

Comment (seisman) [reply CONTRIBUTING.md:191]: > 2. Literal GMT arguments should be **bold** by wrapping the arguments with \*\*
   (two asterisks) on both sides. The argument description should be in *italicized*
   with \* (single asterisk) on both sides.

If I understand it correctly, we all agree that this rule also applies to string-type arguments like **05m**, right? If yes, we probably don't need to add a new rule, but extend this rule by adding one more example.

Comment (core-man) [reply CONTRIBUTING.md]: Add an example in 6af2deb.

**Comment by core-man** [comment]
> 1. Python arguments, such as import statements, Boolean expressions, and function arguments should be wrapped as code by using `` on both sides of the code. Example: \`\`import pygmt\`\` results in `import pygmt`.

I find we use both \`True\` (`True`) and `**True**` (**True**). E.g,

- https://www.pygmt.org/dev/gallery/lines/line_custom_cpt.html?highlight=true
- https://www.pygmt.org/dev/api/generated/pygmt.datasets.load_earth_relief.html?highlight=true

Shall we add an example in the first rule? e.g., \`\`True\`\` results in `True`. This corresponds to Boolean expressions, right?

Meanwhile, shall we move this example in the first rule: \`\`style="v"\`\` results in `style="v"`? This is for function arguments.

**Comment by seisman** [comment]
> Shall we add an example in the first rule? e.g., ``True`` results in `True`. This corresponds to Boolean expressions, right?

Yes.

> Meanwhile, shall we move this example in the first rule: ``style="v"`` results in `style="v"`? This is for function arguments.

Yes.

**Review by seisman — APPROVED** [review]

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -181,11 +181,12 @@ When editing documentation, use the following standards to demonstrate the examp
 
 1. Python arguments, such as import statements, Boolean expressions, and function
    arguments should be wrapped as ``code`` by using \`\` on both sides of the code.
-   Example: \`\`import pygmt\`\` results in ``import pygmt``.
+   Example: \`\`import pygmt\`\` results in ``import pygmt``, \`\`True\`\` results in `True`,
```
Comment (meghanrjones) [inline CONTRIBUTING.md:184]:
```suggestion
   Examples: \`\`import pygmt\`\` results in ``import pygmt``, \`\`True\`\` results in `True`,
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -181,11 +181,12 @@ When editing documentation, use the following standards to demonstrate the examp
...
+    \`\`style="v"\`\` results in `style="v"`.
 2. Literal GMT arguments should be **bold** by wrapping the arguments with \*\*
    (two asterisks) on both sides. The argument description should be in *italicized*
    with \* (single asterisk) on both sides.
-   Example: `**+l**\ *label*` results in **+l***label*.
+   Example: `**+l**\ *label*` results in **+l***label*, `**05m**` results in **05m**.
```
Comment (meghanrjones) [inline CONTRIBUTING.md:189]:
```suggestion
   Examples: `**+l**\ *label*` results in **+l***label*, `**05m**` results in **05m**.
```

**Review by meghanrjones — APPROVED** [review]

**Pull request merged**